crates/net/common/src/dns_node_record_resolve.rs

```diff
@@ -0,0 +1,41 @@
+//! Resolves a [DNSNodeRecord] to a [NodeRecord] by looking up the record in the DNS system.
+
+use reth_network_types::{DNSNodeRecord, NodeRecord};
+use std::io::Error;
+use url::Host;
+
+/// Resolves the host in a [DNSNodeRecord] to an IP address, returning a [NodeRecord]. If the domain
+/// cannot be resolved, an error is returned.
+///
+/// If the host is already an IP address, the [NodeRecord] is returned immediately.
+pub async fn resolve_dns_node_record(node_record: DNSNodeRecord) -> Result<NodeRecord, Error> {
```

Comment:
why is this a standalone function and not a fn of dnsrecord?

Reply:
@mattsse I have moved the fn to be a method of the dns struct in this pr https://github.com/paradigmxyz/reth/pull/8358#pullrequestreview-2077279873

I also think we should consolidate DNSNodeRecord into NodeRecord and perhaps lazily resolve domain names when the tcp/udp getter fns are used